src/main/java/com/openshift/jenkins/plugins/util/ClientCommandRunner.java

```diff
@@ -96,7 +102,44 @@ public void checkRoles(RoleChecker checker) throws SecurityException {
         public Integer invoke(File currentDir, VirtualChannel channel) throws IOException, InterruptedException {
             Proc.LocalProc proc = null;
             try (OutputStream out = this.out; OutputStream err = this.err) {
-                proc = new Proc.LocalProc(command, null, null, out, err, currentDir);
+                // per explanations like https://stackoverflow.com/questions/10035383/setting-the-environment-for-processbuilder
+                // even with propagating updates to the PATH env down to the creations of Proc.LocalProc and ProcessBuilder, 
+                // they don't even effect the environment in which the ProcessBuilder is running. So to find the `oc` when the 
+                // PATH is updated via the 'tool' step, we need to either 
+                // 1) prepend the right dir from the path for 'oc'
+                // 2) invoke a shell that then launches the actual 'oc' command, where we update the PATH prior
+                // Our use of the jenkins durable task and bourne/windows scripts did 2), but because of 
+                // https://bugzilla.redhat.com/show_bug.cgi?id=1625518 we analyze the PATH env var and do 1)
+                String path = envVars.get("PATH");
+                // default if running the openshift jenkins images
+                String dirToUse = "/bin/";
+                if (path == null || path.length() == 0) {
+                    LOGGER.warning("PATH not properly set prior to invocation of 'oc'");
+                } else {
+                    String[] dirs = path.split(File.pathSeparator);
```

Comment:
I remember that there is a short circuit: Linux and Windows searches `PATH` only when the specified command doesn't contain `/` (or `\` on Windows).

Reply:
Interesting ... in our case it should not matter though.  The command passed to this point will always be `oc` or `oc.exe` ... there will be no fully qualified path to oc.